qualtran/resource_counting/_call_graph.py

```diff
@@ -16,14 +16,15 @@
 
 import collections.abc
 from collections import defaultdict
-from typing import Callable, cast, Dict, List, Optional, Sequence, Set, Tuple, Union
+from typing import Callable, cast, Dict, Iterable, List, Optional, Sequence, Set, Tuple, Union
 
 import networkx as nx
 import sympy
 
 from qualtran import Bloq, CompositeBloq, DecomposeNotImplementedError, DecomposeTypeError
 
 BloqCountT = Tuple[Bloq, Union[int, sympy.Expr]]
+BloqCountDictT = Dict[Bloq, Union[int, sympy.Expr]]
```

Comment:
We could perhaps use `Mapping` here instead of `dict`? As `build_call_graph` is an internal function, would make it easier for bloq authors to return any mapping (eg. `Counter` instead of casting to `dict` for mypy).

Reply:
Good idea, done.